### ISSUE TITLE
Start and shutdown scripts issue with currentpid file

### ISSUE THREAD

**Issue by ggalves** [azkaban/azkaban#1748]
Here's the problem: 
When running start script, it writes the PID of Azkaban on `currentpid` file. Shutdown script uses this file to stop Azkaban.
But if I run start script again before shutting it down, start will fail as there's already an instance of Azkaban running, but`currentpid` file will be overwritten with this new PID, losing the old one (that is the right one). 
After this, I'm not able to use shutdown script anymore, as it tries to kill a process that already failed.

**Comment by reallocf** [comment]
Yes, this is an issue. What I tend to do when this happens is run `ps aux | grep azkaban` to find the running server, then `kill` manually. Does this not work for you?

**Comment by ggalves** [comment]
Yes, it does. But I requested someone to restart azkaban and he had this problem and thought he had restarted (did not see the output). Things continued not to work. Took me some time to realize that the server did not restarted.

Not a major issue, just pointing as I think this should improve.

Thanks!

**Comment by reallocf** [comment]
Sounds like something worth fixing to me as long as it doesn't introduce a ton of new complexity. Thanks for bringing it up!

**Comment by sarsjits** [comment]
What is a way to shutdown the executor gracefully? Because the shutdown script simply kills the process, failing all the running and the preparing jobs assigned to that executor.
